### PR TITLE
[Shop][Behat] Upgraded EmailChecker and fixed verification on registration

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Hook/SpoolContext.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * @var Filesystem
+     */
+    private $fileSystem;
+
+    /**
+     * @param EmailCheckerInterface $emailChecker
+     * @param Filesystem $fileSystem
+     */
+    public function __construct(EmailCheckerInterface $emailChecker, FileSystem $fileSystem)
```
Comment (pamil) [inline src/Sylius/Behat/Context/Hook/SpoolContext.php]: It's `Filesystem`, not `FileSystem`, same for variable name.

File: src/Sylius/Behat/Context/Hook/SpoolContext.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $this->spoolDirectory = $emailChecker->getSpoolDirectory();
+        $this->fileSystem = $fileSystem;
+    }
+
+    /**
+     * @BeforeScenario @email
```
Comment (pamil) [inline src/Sylius/Behat/Context/Hook/SpoolContext.php]: Nice, does it really works like it should? :tada:

Comment (lchrusciel) [inline src/Sylius/Behat/Context/Hook/SpoolContext.php]: Nice!

File: features/account/verification/email_verification.feature [hunk 3]
```diff
@@ -44,10 +44,10 @@ Feature: Verifying account's email address
...
         Then I should be unable to resend the verification email
 
-    @ui
+    @ui @email
     Scenario: Receiving account verification email after registration
         When I register with email "ghastly@bespoke.com" and password "suitsarelife"
         Then I should be logged in
         And I should be notified that my account has been created and the verification email has been sent
-        And the verification email should be sent to "ghastly@bespoke.com"
+        And "ghastly@bespoke.com" should receive 2 email
```
Comment (lchrusciel) [inline features/account/verification/email_verification.feature]: `And 2 emails should be sent to ghastly@bespoke.com"` ?

Comment (pamil) [inline features/account/verification/email_verification.feature]: Ghastly should receive them too! 👯

File: src/Sylius/Behat/Context/Hook/SpoolContext.php (hunk 2 above)
Comment (NeverResponse) [inline src/Sylius/Behat/Context/Hook/SpoolContext.php]: Only if you tag the Feature/Scenario and the Hook is in configuration, but yeah works like a charm 😃

File: src/Sylius/Behat/Context/Hook/SpoolContext.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+namespace Sylius\Behat\Context\Hook;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Test\Services\EmailCheckerInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class SpoolContext implements Context
```
Comment (pjedrzejewski) [inline src/Sylius/Behat/Context/Hook/SpoolContext.php]: EmailSpoolContext?

**Comment by pjedrzejewski** [comment]
Really cool Jasiek, thanks!

**Pull request merged**